### PR TITLE
Add rancher deployment scripts

### PR DESCRIPTION
**Pull request by adamgraves-choices**
- One for the deployment of PRs, creating a domain prefix for the services
- One to deploy staging, always using the master branch
- One to upgrade an individual "Service" in a stack

### CODE REVIEW

File: scripts/rancher_single_service_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def main(argv):
+    environmentName = ''
+    serviceName = ''
+    newImage = ''
+    newImageTag = 'latest'
+
+    try:
+        opts, args = getopt.getopt(argv, "hs:i:e:t:", ["service-name=", "image-name=", "environment=", "image-tag="])
```
Comment (neilbmclaughlin) [inline scripts/rancher_single_service_upgrade.py]: If we need to stick with the python code we will need to add a python linter.

**Comment by neilbmclaughlin** [comment]
[Change deployment mechanism from Azure WebApps to Docker](https://trello.com/c/F4ySZOIz/685-change-deployment-mechanism-from-azure-webapps-to-docker)

**Comment by st3v3nhunt** [comment]
[All developers able to make code changes and test them using docker.](https://trello.com/c/lXDzR8a0/732-all-developers-able-to-make-code-changes-and-test-them-using-docker)

**Comment by neilbmclaughlin** [comment]
Rebase please so I can review & approve.

**Pull request merged**